File: level1JS/guessOfCollatz.js
function solution(num) {
    let answer = 0;
    let target = num;
    for(let i = 0; i < 500; i++) {
        if(target === 1) {
            answer = i;
            break;
        }

        target = target % 2 === 0 ? target / 2 : target * 3 + 1;
    }
    return target !== 1 ? -1 : answer;
}

console.log(solution((16)));

File: level1JAVA/src/GuessOfCollatz.java
class GuessOfCollatz {
    public int solution(int num) {
        long result = num;
        for(int i = 0; i <= 500; i++) {
            if(result == 1) return i;
            result = result % 2 == 0 ? result / 2 : result * 3 + 1;
        }
        return -1;
    }

    public static void main(String[] args) {
        GuessOfCollatz guessOfCollatz = new GuessOfCollatz();
        System.out.println("result = " + guessOfCollatz.solution(626331));
    }
}
